src/react-components/src/components/Template/Template.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from "react";
+import bem from "../../utils/bem";
+import { LayoutTypes } from "./LayoutTypes";
+import Placeholder from "../00-base/Placeholder";
+
+export interface TemplateProps {
+  /** Enum to control if Template has left/right sidebar or is full width. */
+  layoutType: LayoutTypes;
+
+  /** Content intended for inside <header>. */
+  headerContent: JSX.Element;
+
+  /** Primary content within <main> element. */
+  primaryContent: JSX.Element;
+
+  /** Content intended for inside <footer>. */
+  footerContent: JSX.Element;
+
+  /** Heroes appear inside <main> but extend full-width of template. */
+  heroContent?: JSX.Element;
+
+  /** Separate block for content above primary content within <main>. */
+  topContent?: JSX.Element;
+
+  /** Content intended for inside the sidebar within <main>. */
+  sidebarContent?: JSX.Element;
+}
+
+export default function Template(props: TemplateProps) {
+  let {
+    layoutType,
+    headerContent,
+    heroContent,
+    topContent,
+    sidebarContent,
+    primaryContent,
+    footerContent,
+  } = props;
+
+  const templateBaseClass = "layout-container";
+
+  let modifiers = [];
+  let mainModifiers = [];
+  let leftSidebar;
+  let rightSidebar;
+
+  const createSidebar = (sidebarContent: JSX.Element) => (
+    <div className={bem("content-secondary", modifiers)}>{sidebarContent}</div>
+  );
+
+  if (layoutType === LayoutTypes.SidebarLeft) {
+    modifiers.push("with-sidebar-left");
+    leftSidebar = createSidebar(sidebarContent);
+  } else if (layoutType === LayoutTypes.SidebarRight) {
+    modifiers.push("with-sidebar-right");
+    rightSidebar = createSidebar(sidebarContent);
+  } else {
+    return;
+  }
+
+  if (layoutType === LayoutTypes.SidebarLeft || LayoutTypes.SidebarRight) {
+    mainModifiers.push("with-sidebar");
+  }
+
+  return (
```

Comment:
I ... think this might cause problems.  In ResearchNow we don't build our pages out with a `header`, `main`, and `footer` on the same page -  we inject the header with the script (https://github.com/NYPL/sfr-bookfinder-front-end/blob/development/src/views/index.ejs#L61) 

This means that we can't use `<Template>` as a component, but we can still make use of all the CSS by calling the classnames directly, using this as an example.  Is that OK, or is this approach something we should re-think?

Reply:
I think maybe this component could exist as an example of how you're supposed to build your pages, with the right semantic elements and classes, rather than something that a consumer is supposed to import and use.

Reply:
Reno does the same thing, it just does the script injection inside the `header` (so a few lines up from where you have it). Is the issue that I've set this up so the content blocks only take JSX.Elements? Would it be solved if they could take just some compiled HTML?